test/dark-gravity-wave-test.js

```diff
@@ -1,126 +1,286 @@
+            const highTargetBits = 0x1b0777d4; // 453474260
+            const result = dgw.isValidTarget(highTargetBits, blocks);
+            expect(result).to.equal(true);
+        });
+
+        it('should be invalid for too high target', () => {
+            const highTargetBits = 0x208fffff; // 546308095
+            const result = dgw.isValidTarget(highTargetBits, blocks);
+            expect(result).to.equal(false);
+        });
+
+        it('should be valid for the max regtest target if less than 25 blocks has been mined', () => {
+            const highTargetBits = 0x207fffff;
+            const result = dgw.isValidTarget(highTargetBits, blocks.slice(0, 10));
+            expect(result).to.equal(true);
+        });
+
+                it('should be valid for int headerBits', () => {
+            const highTargetInt = 1000;
+            const result = dgw.isValidTarget(highTargetInt, blocks);
+            expect(result).to.equal(true);
+        });
+
+        it('should be invalid maxTarget = 0x207fffff', () => {
+            const highTargetInt = 0x207fffff;
+            const result = dgw.isValidTarget(highTargetInt, blocks);
+            expect(result).to.equal(false);
+        });
+
+        it('should be valid when num blocks < maxBlocks and maxTarget = 0x207fffff', () => {//TODO is it True?
+            const highTargetInt = 0x207fffff;
+            const result = dgw.isValidTarget(highTargetInt, blocks.slice(0, 10));
+            expect(result).to.equal(true);// TODO change on false?
+        });
+
+        it('should be invalid maxTarget = 0x207fffff as int', () => {
+            const highTargetInt = 545259519;
+            const result = dgw.isValidTarget(highTargetInt, blocks);
+            expect(result).to.equal(false);
+        });
+
+        it('should be valid for maxTarget > 0x207fffff', () => {
+            const highTargetInt = 0x20800000;
+            const result = dgw.isValidTarget(highTargetInt, blocks);
+            expect(result).to.equal(true);
+        });
+
+        it('should be valid for maxTarget > 0x207fffff as int', () => {
+            const highTargetInt = 0x20800000;
+            const result = dgw.isValidTarget(highTargetInt, blocks);
+            expect(result).to.equal(true);
+        });
+
+        it('should be valid when allHeaders is empty array', () => {
+            const highTargetInt = 0x20800000;
+            const result = dgw.isValidTarget(highTargetInt, [{}]);
+            expect(result).to.equal(true);
+        });
+
+        it('should throw error for allHeaders is {}', () => {
+            const highTargetBits = 0x208fffff;
+            expect(() => {
+                dgw.isValidTarget(highTargetBits, {});
+            }).to.throw('allHeaders.slice is not a function');
+        });
+
+        it('should throw error for allHeaders is string', () => {
+            // TODO not very good that we allow string although it's length < 25 and ignored in getTarget
```

Comment:
yes agree.  maybe we should add some validation on the `allHeaders` param provided?